data/src/main/java/com/linkedin/data/schema/validation/ValidateDataAgainstSchema.java

```diff
@@ -526,6 +526,10 @@ else if (_recursive)
             validate(memberElement, memberSchema, value);
           }
         }
+        else if (!schema.isAllowEmptyUnionResponse())
+        {
+          addMessage(element, "DataMap should have at least one entry for a union type or specify the member in projection");
```

Comment:
You can remove the part about "specify member in projection". It can cause confusion about this error, which if happens means the resource method didn't populate the union field correctly. Using a projection to avoid the error is not the correct way to handle it.